docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,223 @@
+
+`\\.\C:\Test\Foo.txt`  
+`\\?\C:\Test\Foo.txt`
+
+> [!NOTE]
+> DOS device path syntax is supported on .NET implementations running on Windows starting with .NET Core 1.1 and .NET Framework 4.6.2.
+
+The DOS device path consists of the following components:
+
+- The device path specifier (`\\.\` or `\\?\`), which identifies the path as a DOS device path.
+- A symbolic link to the "real" device object (C: in this case).
+
+   The first segment of the DOS device path after the device path specifier identifies the volume or drive. (For example, `\\?\C:\` and `\\.\BootPartition\`.)
+
+   There is a specific link for UNCs that is called, not surprisingly, `UNC`. For example:
+
+      `\\.\UNC\Server\Share\Test\Foo.txt`
+      `\\?\UNC\Server\Share\Test\Foo.txt`
+
+    For device UNCs, the server/share portion is forms the volume. For example, in `\\?\server1\e:\utilities\\filecomparer\`, the server/share portion is server1\utilities. This is significant when calling a method such as <xref:System.IO.Path.GetFullPath(String,String)?displayProperty=nameWithType> with relative directory segments; it is never possible to navigate past the volume. 
+
+DOS device paths are fully qualified by definition. Relative directory segments (`.` and `..`) are not allowed. Current directories never enter into their usage.
+
+## Example: Ways to refer to the same file
+
+The following example illustrates some of the ways in which you can refer to a file when using the APIs in the <xref:System.IO> namespace. The example instantiates a <xref:System.IO.FileInfo> object and uses its <xref:System.IO.FileInfo.Name> and <xref:System.IO.FileInfo.Length> properties to display the filename and the length of the file.
+
+[!code-csharp[referring-to-the-same-file](~/samples/snippets/standard/io/file-names/cs/file-refs.cs)]
+[!code-vb[referring-to-the-same-file](~/samples/snippets/standard/io/file-names/vb/file-refs.vb)]
+
+## Path normalization
+
+Almost all paths passed to Windows APIs are normalized. During normalization, Windows performs the following steps:
+
+- Identifies the path.
+- Applies the current directory to partially qualified (relative) paths.
+- Canonicalizes component and directory separators.
+- Evaluates relative directory components (`.` for the current directory and `..` for the parent directory).
+- Trims certain characters.
+
+This normalization happens implicitly, but you can do it explicitly by calling the Windows [GetFullPathName() function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx) using P/Invokle. You can also call the <xref:System.IO.Path.GetFullPath%2A?displayProperty=nameWithType> method, which wraps a call to the  [GetFullPathName() function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx).
```

Comment:
Spelling for P/Invoke. I'd consider flipping this to encourage using Path.GetFullPath() as the "normal" .NET way to do this. Something like ", but you can do it explicitly by calling the Path.GetFullPath() method. This method wraps the Windows GetFullPathName() function."